### PR TITLE
Adds the `.is_pinned` mech

### PR DESCRIPTION
**Pull request by heypr**
- Requested by [zemenus](https://discord.com/channels/315163488085475337/1230995395765600297)

### CODE REVIEW

File: src/main/java/com/denizenscript/ddiscordbot/objects/DiscordMessageTag.java [hunk 1]
```diff
@@ -437,6 +437,24 @@ public static void register() {
             return result;
         });
 
+        // <--[mechanism]
+        // @object DiscordMessageTag
+        // @name crosspost
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/ddiscordbot/objects/DiscordMessageTag.java]: what

File: src/main/java/com/denizenscript/ddiscordbot/objects/DiscordMessageTag.java [hunk 2]
```diff
@@ -471,6 +471,30 @@ public static void register() {
...
+        // @tags
+        // <DiscordMessageTag.is_pinned>
+        // -->
+        tagProcessor.registerMechanism("is_pinned", false, ElementTag.class, (object, mechanism, input) -> {
+            boolean pinned = input.asBoolean();
+            Message message = object.getMessage();
+            try {
+                if (pinned) {
+                    message.pin().submit();
+                } else {
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/ddiscordbot/objects/DiscordMessageTag.java]: newline between `}` and `else`

File: src/main/java/com/denizenscript/ddiscordbot/objects/DiscordMessageTag.java [hunk 3]
```diff
@@ -471,6 +471,31 @@ public static void register() {
...
         });
+        
+        // <--[mechanism]
+        // @object DiscordMessageTag
+        // @name is_pinned
+        // @input ElementTag(Boolean)
+        // @description
+        // Sets whether the message is pinned.
+        // @tags
+        // <DiscordMessageTag.is_pinned>
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/ddiscordbot/objects/DiscordMessageTag.java:483]: Oh, also: the relevant tag meta needs to be updated to link the mech

File: src/main/java/com/denizenscript/ddiscordbot/objects/DiscordMessageTag.java [hunk 4]
```diff
@@ -471,6 +472,31 @@ public static void register() {
...
+        // -->
+        tagProcessor.registerMechanism("is_pinned", false, ElementTag.class, (object, mechanism, input) -> {
+            boolean pinned = input.asBoolean();
+            Message message = object.getMessage();
+            try {
+                if (pinned) {
+                    message.pin().submit();
+                }
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/ddiscordbot/objects/DiscordMessageTag.java]: stray space added to end of line

**Review by mcmonkey4eva — APPROVED** [review]

File: src/main/java/com/denizenscript/ddiscordbot/objects/DiscordMessageTag.java [hunk 5]
```diff
@@ -471,6 +472,31 @@ public static void register() {
...
+        // @name is_pinned
+        // @input ElementTag(Boolean)
+        // @description
+        // Sets whether the message is pinned.
+        // @tags
+        // <DiscordMessageTag.is_pinned>
+        // -->
+        tagProcessor.registerMechanism("is_pinned", false, ElementTag.class, (object, mechanism, input) -> {
+            boolean pinned = input.asBoolean();
+            Message message = object.getMessage();
```
Comment (tal5) [inline src/main/java/com/denizenscript/ddiscordbot/objects/DiscordMessageTag.java]: Missing `Mechanism#requireBoolean`

File: src/main/java/com/denizenscript/ddiscordbot/objects/DiscordMessageTag.java [hunk 6]
```diff
@@ -471,6 +472,34 @@ public static void register() {
...
+            if (!mechanism.requireBoolean()) {
+                return;
+            }
```
Comment (tal5) [inline src/main/java/com/denizenscript/ddiscordbot/objects/DiscordMessageTag.java:488]: The check should come _before_ converting the input (should basically always be the first thing)

**Review by tal5 — APPROVED** [review]

**Pull request merged**